web/views.py

```diff
@@ -50,6 +53,23 @@ def coaches(request):
     return render(request, 'web/coaches.html')
 
 
+def records(request):
+    results = SwimRecord.objects.all()
+    team_list = list()
+    pool_list = list()
+    for elem in results:
+        if elem.record_type == SwimRecord.RECORD_TYPE_TEAM:
+            elem.pool_verbose = elem.get_pool_display()
+            team_list.append(elem)
+        else:
+            elem.swimmer_team_verbose = elem.get_swimmer_team_display()
+            pool_list.append(elem)
```

Comment:
This is starting to muddle the responsibilities of the *view* and the *controller*. It's the controller's job to separate the data (the model). It's the view's job to determine what is "verbose". In general, the model should be considered an **immutable** object representation of the data.

Lines 58-66 can be succinctly achieved via list comprehensions for both clarity and speed.

```python
team_list = [e for e in results if e.record_type == SwimRecord.RECORD_TYPE_TEAM]
pool_list = [e for e in results if e.record_type == SwimRecord.RECORD_TYPE_POOL]
```
Then, in the view, you can choose to display the verbose name:

```html
<td><abbr title="{{ instance.get_pool_verbose }}">{{ instance.pool }}</abbr></td>
```